apollo-client/src/main/java/com/ctrip/framework/apollo/spring/annotation/ApolloAnnotationProcessor.java

```diff
@@ -1,89 +1,138 @@
+        return bean;
+    }
 
-  @Override
-  public Object postProcessAfterInitialization(Object bean, String beanName) throws BeansException {
-    return bean;
-  }
+    @Override
+    public Object postProcessAfterInitialization(Object bean, String beanName) throws BeansException {
+        return bean;
+    }
 
-  private void processFields(Object bean, Field[] declaredFields) {
-    for (Field field : declaredFields) {
-      ApolloConfig annotation = AnnotationUtils.getAnnotation(field, ApolloConfig.class);
-      if (annotation == null) {
-        continue;
-      }
+    private void processFields(Object bean, Field[] declaredFields) {
+        for (Field field : declaredFields) {
+            ApolloConfig annotation = AnnotationUtils.getAnnotation(field, ApolloConfig.class);
+            if (annotation == null) {
+                continue;
+            }
 
-      Preconditions.checkArgument(Config.class.isAssignableFrom(field.getType()),
-          "Invalid type: %s for field: %s, should be Config", field.getType(), field);
+            Preconditions.checkArgument(Config.class.isAssignableFrom(field.getType()),
+                    "Invalid type: %s for field: %s, should be Config", field.getType(), field);
 
-      String namespace = annotation.value();
-      Config config = ConfigService.getConfig(namespace);
+            String namespace = annotation.value();
+            Config config = ConfigService.getConfig(namespace);
 
-      ReflectionUtils.makeAccessible(field);
-      ReflectionUtils.setField(field, bean, config);
+            ReflectionUtils.makeAccessible(field);
+            ReflectionUtils.setField(field, bean, config);
+        }
     }
-  }
 
-  private void processMethods(final Object bean, Method[] declaredMethods) {
-    for (final Method method : declaredMethods) {
-      ApolloConfigChangeListener annotation = AnnotationUtils.findAnnotation(method, ApolloConfigChangeListener.class);
-      if (annotation == null) {
-        continue;
-      }
+    private void processAutoRefreshFields(final Object bean, Field[] declaredFields) {
+        for (final Field field : declaredFields) {
+            EnableAutoResfresh annotation = AnnotationUtils.getAnnotation(field, EnableAutoResfresh.class);
+            if (annotation == null) {
+                continue;
+            }
+            String namespace = annotation.value();
+            final Config config = ConfigService.getConfig(namespace);
+            config.addAutoChangeListener(new AutoConfigChangeListener() {
+                @Override
+                public void autoChange(ConfigChangeEvent changeEvent) {
+                    ReflectionUtils.makeAccessible(field);
+                    //If it is the processing of complex data types, I suggest not to do AutoRefresh here or to handle it yourself
+                    //add by 258737400@qq.com 2017-12-18
+                    String type = field.getGenericType().toString();
```

Comment:
Remember we should always use Spring's `environment.getProperty()` to get the property value because the key might be overridden in a different hierarchy.

Reply:
What do you mean by modifying the code here?